examples/vue-code-file/eslint.config.js

```diff
@@ -1,18 +1,31 @@
-import vueParser from 'vue-eslint-parser';
+import { mergeProcessors } from 'eslint-merge-processors';
+import pluginVue from 'eslint-plugin-vue';
+import processorVueBlocks from 'eslint-processor-vue-blocks';
 import js from '@eslint/js';
 import graphqlPlugin from '@graphql-eslint/eslint-plugin';
 
 export default [
   {
-    files: ['**/*.js', '**/*.vue'],
+    files: ['**/*.js'],
     processor: graphqlPlugin.processor,
     rules: js.configs.recommended.rules,
   },
+  ...pluginVue.configs['flat/recommended'],
   {
     files: ['**/*.vue'],
-    languageOptions: {
-      parser: vueParser,
-    },
+    // `eslint-plugin-vue` will set a default processor for `.vue` files
+    // we use `eslint-merge-processors` to extend it
+    processor: mergeProcessors([
+      pluginVue.processors['.vue'],
+      processorVueBlocks({
+        blocks: {
+          styles: false,
+          customBlocks: true,
+          script: true,
```

Comment:
also what about this `scriptSetup`

https://github.com/antfu/eslint-processor-vue-blocks/blob/9826f22e4180d95adda2edd1585e65d248e6577b/src/index.ts#L69-L74

Reply:
Script setup is essentially the new standard, best practice way to write Vue components. Almost all my code is in that, or migrating to it. Should certainly be supported.